### PR TITLE
[POC] Jlayout for list field

### PR DESCRIPTION
**Pull request by degobbis**
### Summary of Changes
Code adapted for the use of JLayout. The field can than be overwritten using a layout file.
cc @zero-24 

**Addition:**  
If we assume this PR, we can remove the overrides in all fields that depends on this one.  
They all then use the layout renderer method of this field.  

I will gladly make the adjustments of dependend fields in a separate PR if this change is accepted.

### Testing Instructions
Go into Joomla backend and select Menus->Main menu

### Actual result BEFORE applying this Pull Request
The list field for filtering (or each other list field), works and looks as expected.

listFilds

### Expected result AFTER applying this Pull Request
The list field for filtering (or each other list field), works and looks as expected.  
Nothing breaks.

### Documentation Changes Required
None.

### CODE REVIEW

**Comment by ChristineWk** [comment]
I have tested this item :white_check_mark: successfully on b43f6af624c978be4b02e2d73e3b2da3f0406c08<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29930">issues.joomla.org/tracker/joomla-cms/29930</a>.</sub>

@degobbis tried some filter after selecting: Menus->Main menu (they were OK) - don't know which you mean exactly.

**Comment by degobbis** [comment]
> 
> 
> @degobbis tried some filter after selecting: Menus->Main menu (they were OK) - don't know which you mean exactly.

@ChristineWk Thx for testing.
For a better understanding I have extended the test instruction a little

**Comment by Fedik** [comment]
J4 already have it, I hope you copied this one from J4 https://github.com/joomla/joomla-cms/blob/4.0-dev/layouts/joomla/form/field/list.php :)
Then this is okay

**Comment by degobbis** [comment]
Not exactly, because the layout in J4 does not quite match the processing of the field in J3.  
But I have transferred the generation from the list field into a separate layout file.

**Comment by degobbis** [comment]
The trevis-ci test crashes on this line:  
https://github.com/joomla/joomla-cms/blob/3.10-dev/libraries/joomla/session/handler/native.php#L235

Comming from here:  
https://github.com/joomla/joomla-cms/blob/3.10-dev/libraries/cms/html/category.php#L48

Startet from this call:  
https://github.com/joomla/joomla-cms/blob/3.10-dev/libraries/legacy/form/field/category.php#L65

When I tested it on the staging or 3.10-dev branch, without a PR, it also crashes on the same line.
Is someone here that has an idea how to fix it?

**Comment by TheWhykiki** [comment]
I have tested this item :white_check_mark: successfully on 0ed76e6db367097931fafa16cfe34dd79ce7c122<br /><br />No problems found, works fine.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/29930">issues.joomla.org/tracker/joomla-cms/29930</a>.</sub>

File: layouts/joomla/form/field/list.php [hunk 1]
```diff
@@ -0,0 +1,103 @@
+<?php
+/**
+ * @package     Joomla.Site
+ * @subpackage  Layout
+ *
+ * @copyright   Copyright (C) 2005 - 2020 Open Source Matters, Inc. All rights reserved.
```
Comment (PhilETaylor) [inline layouts/joomla/form/field/list.php]:
```suggestion
 * @copyright   (C) 2021 Open Source Matters, Inc. <https://www.joomla.org>
```

**Comment by zero-24** [comment]
Hi @degobbis I'm sorry that this has never been merged into the J3 series. While I was involved in the initial thought process building up to this PR. I agree now that this PR should not be merged into J3. I understand and remember our thoughts but I'm not intending to change the behavior now with the last regular release of J3

Thanks for the PR and thanks for understanding

**Pull request closed without merging**